intro/setup/index.html

```diff
@@ -0,0 +1,349 @@
+    </p>
+
+    <p>
+        Next up, after importing pygame, the first thing you do is to <i>always</i> initialize 
+        the module. If you don't do this step it can lead to problems later on, this initializes 
+        many submodules such as pygame.font, pygame.mixer, etc. which we will cover in detail later.
+{% highlight python %}
+import pygame
+
+pygame.init()
+{% endhighlight %}
+    </p>
+
+    <p>
+        Now let's actually open up a window, for this, we will look at the
+         <a href="https://www.pygame.org/docs/ref/display.html#pygame.display.set_mode">pygame.display.set_mode</a> function. 
+        This allows us to initialize the window, and it returns a pygame.Surface when called. A 
+        pygame.Surface is a data type which allows us to manipulate and draw pixels on it, this special 
+        surface is directly linked to what shows up on the window, so everything we want to appear 
+        on the window should be drawn on this special surface first. For now, let us look at what we need in 
+        order to call pygame.display.set_mode
+
+{% highlight python %}
+import pygame
+
+pygame.init()
+
+
+WIN_SIZE = (420, 250)
+screen = pygame.display.set_mode(WIN_SIZE)
+{% endhighlight %}
+
+        Notice that we passed in a tuple containing two integers, the first integer is the <i>width</i> 
+        of the window we want to create, and the second is the <i>height</i> of the window. This fills in the size 
+        parameter which is the only required argument to call this function.
+    </p>
+
+    <p>
+        When you run this code, the window opens up and then closes immediately. Why is that? 
+        This is because there is nothing telling pygame that the window is alive, nor is the 
+        python program continuing to run. After creating that special surface we were talking about 
+        it just closes. To keep the window alive and running, we need to keep the Python program running. 
+        To do this, let us create a while loop. From this point onward, the code presented is assumed to be added at the end 
+        of the file unless mentioned otherwise. 
+
+{% highlight python %}
+while True:
+    pass
+{% endhighlight %}
+
+        We notice two problems here:
+        <ol>
+            <li>The window is really laggy, it hangs immediately. To close the application run ctrl + c in your terminal.</li>
+            <li>The window is small, a bit too small.</li>
+        </ol>
+
+        We will address both the issues present here. 
+    </p>
+
+    <h3>The window is really laggy.</h3>
+    <p>
+        While the python program is still running, pygame has no way of understanding 
+        that the window is still running, it needs some kind of system that checks for these events.
+        For this, we can grab all the window events that is going on
+
+{% highlight python %}
+while True:
+    events = pygame.event.get()
+{% endhighlight %}
+
+        This should make it responsive, and not hang. 
```

Comment:
Well, it still kinda hangs because there's really no difference just by grabbing the events. Still have to close the window by force-killing the script (as I now see you mention later), but I could imagine someone trying it out at this point before reading further and getting confused here when it doesn't work

Reply:
No it does not hang, the X button still wouldnt work because the behaviour to quit the game hasnt been defined